torch/_inductor/codegen/cpp.py

```diff
@@ -3132,6 +3128,149 @@ def to_dtype(x, dtype, src_dtype=None, use_compute_types=True):
         return self
 
 
+def get_loop_body_lowp_fp(_body: ir.LoopBody):
+    sub_blocks = [_body.root_block] + list(_body.subblocks.values())
+
+    _lowp_fp_type: Optional[torch.dtype] = None
+
+    for sub_block in sub_blocks:
+        for _node in sub_block.graph.nodes:
+            # TODO(Eikan): Regarding get_index and index_expr, we should conclude the
+            # the data type as well.
+            if _node.op == "placeholder" or _node.target in (
+                "get_index",
+                "index_expr",
+            ):
+                continue
+
+            # Fast path if all operations can support bf16/fp16 without converting to fp32
+            if _node.target not in [
+                "load",
+                "store",
+                "abs",
+                "neg",
+                "output",
+            ]:
+                return None
```

Comment:
Add the comment for this function. It will return `bf16/fp16` when all the nodes inside loopbody can codegen in `bf16/fp16` without casting to `fp32`, otherwise returns None.